### PR TITLE
Deprecate IFluidConfiguration and all usages, remove configuration from ContainerRuntime

### PR DESCRIPTION
**Pull request by ChumpChief**
First part of #3854 

In this change, I deprecate everything and go ahead with removal from the `ContainerRuntime` since I'm not seeing current usage outside of smde (which uses it incorrectly anyway, failing to refresh the view on reconnect).

Based on smde, it looks like it was initially implemented to provide UI hints to `IFluidHTMLView`s.  We don't recommend that interface any longer, and a well-written view will likely have access to more appropriate objects to discover connection characteristics (e.g. consider the external-views type examples where the host has access to the container).  When combined with the lack of usage, it doesn't seem necessary to supply a replacement IMO.

### CODE REVIEW

File: BREAKING.md [hunk 1]
```diff
@@ -28,6 +29,12 @@ The `getQuorum()` method on `IContainer` and the `quorum` member of `IContainerC
 ### `IChannel.summarize` split into sync and async
 `IChannel` now has two summarization methods instead of a single synchronous `summarize`. `getAttachSummary` is synchronous to prevent channel modifications during summarization, `summarize` is asynchronous.
 
+### `IFluidConfiguration` deprecated and `configuration` member removed from `ContainerRuntime`
+
+The `IFluidConfiguration` interface has been deprecated and will be removed in an upcoming release.  This will include removal of the `configuration` member of the `IContainerContext` at that time.  To inspect whether the document is in readonly state, you should instead query `container.readOnlyInfo.readonly`.
```
Comment (heliocliu) [inline BREAKING.md:34]: can you add which packages the affected APIs are from?

File: BREAKING.md [hunk 2]
```diff
@@ -28,6 +29,12 @@ The `getQuorum()` method on `IContainer` and the `quorum` member of `IContainerC
...
+
+The `configuration` member of `ContainerRuntime` has also been removed.
```
Comment (heliocliu) [inline BREAKING.md:36]: Should this be `FluidConfiguration`?

**Review by heliocliu — APPROVED** [review]

File: BREAKING.md (hunk 1 above)
Comment (ChumpChief) [reply BREAKING.md]: Added in latest

File: BREAKING.md (hunk 2 above)
Comment (ChumpChief) [reply BREAKING.md]: Good catch, fixed in latest :)

**Pull request merged**